internal/pkg/ccloudv2/billing.go

```diff
@@ -0,0 +1,53 @@
+package ccloudv2
+
+import (
+	"context"
+	billing "github.com/confluentinc/ccloud-sdk-go-v2/billing/v1"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"net/http"
+)
+
+func newBillingClient(url, userAgent string, unsafeTrace bool) *billing.APIClient {
+	cfg := billing.NewConfiguration()
+	cfg.Debug = unsafeTrace
+	cfg.HTTPClient = newRetryableHttpClient(unsafeTrace)
+	cfg.Servers = billing.ServerConfigurations{{URL: url}}
+	cfg.UserAgent = userAgent
+
+	return billing.NewAPIClient(cfg)
+}
+
+func (c *Client) billingApiContext() context.Context {
+	return context.WithValue(context.Background(), billing.ContextAccessToken, c.AuthToken)
+}
+
+func (c *Client) ListBillingCosts(startDate, endDate string) ([]billing.BillingV1Cost, error) {
+	var list []billing.BillingV1Cost
+
+	done := false
+	pageToken := ""
+	for !done {
+		page, httpResp, err := c.executeListCosts(startDate, endDate, pageToken)
+		if err != nil {
+			return nil, errors.CatchCCloudV2Error(err, httpResp)
+		}
+		list = append(list, page.GetData()...)
+
+		pageToken, done, err = extractNextPageToken(page.GetMetadata().Next)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return list, nil
+}
+
+func (c *Client) executeListCosts(startDate, endDate, pageToken string) (billing.BillingV1CostList, *http.Response, error) {
+	req := c.BillingClient.CostsBillingV1Api.ListBillingV1Costs(c.billingApiContext()).
+		PageSize(10000).
```

Comment:
Are you sure you want this to be so big?

Reply:
yes, larger page size would be better here since our largest customers have over 10000 line items in a month.

Reply:
Oh wow, does it take a while to fetch when there are that many records?

Reply:
We've seen the largest orgs (1 or 2 which are larger than everyone else) may take some time in the seconds, but on average it's still sub-second.
https://ccloud-production.datadoghq.com/apm/resource/cc-billing-service/opencensus/890b3a4ef53050d6?query=%40_top_level%3A1%20env%3Aprod%20service%3Acc-billing-service%20operation_name%3Aopencensus%20resource_name%3A%22GET%20%2Fbilling%2Fv1%2Fcosts%22&env=prod&spanType=service-entry&topGraphs=latency%3Alatency%2CbreakdownAs%3Apercentage%2Cerrors%3Acount%2Chits%3Acount&traces=qson%3A%28data%3A%28%29%2Cversion%3A%210%29&start=1683218550911&end=1683823350911&paused=false

Reply:
Ok, this will be highest output-producing command, so we should test that printing 10k+ rows doesn't cause the terminal to stall.

Reply:
@brianstrauch  I can do this in the integration tests.

Reply:
I was more interested in seeing if large amounts of data would cause the terminal to freeze or crash (in practice, as opposed to integration tests)

Reply:
+1 on this one

Reply:
Another thing we might want to consider here is scrollback limits on user terminals. 10,000 lines should be within the default for most users at this point (default should be based on available memory), but it would still be annoying if a user had to change their terminal settings to see this output.

User-side pagination (let user tab through pages of results) is on our radar, and it might alleviate both mine and Brian's concerns.